packages/core/src/typings/openapi.ts

```diff
@@ -112,63 +119,84 @@ export interface Oas3Xml {
 }
 
 // common fields for OpenAPI Schema v3.x
-interface Oas3XSchemaBase<T> {
+interface Oas3XSchemaBase<T extends Oas3Schema | Oas3_1Schema> {
   $ref?: string;
-  properties?: { [name: string]: T };
-  additionalProperties?: boolean | T;
+  properties?: { [name: string]: Referenced<T> };
+  additionalProperties?: boolean | { [name: string]: Referenced<T> };
```

Comment:
Oh, one more thing. I guess it should be just `T` as `additionalProperties` accepts a schema:

```suggestion
  additionalProperties?: boolean | T;
```